### PR TITLE
organized tests, cleanup of comments, marked some api's as return of null is possible

### PR DESCRIPTION
**Pull request by brentschmaltz**
moved and cleaned up tests under DownstreamRestApiTest to M.I.Abstractions.Tests
each test in its own namespace
cleaned up some comments
added nuget.config
moved Directory.Build.Props to root folder, test has its own Directory.Build.Props

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
```
Comment (jmprieur) [inline NuGet.config:5]: What is it for?

**Review by jmprieur — APPROVED** [review]
Thanks @brentschmaltz 
LGTM

File: NuGet.config (hunk 1 above)
Comment (brentschmaltz) [reply NuGet.config:5]: Standard nuget.config that Wilson uses.

**Comment by brentschmaltz** [comment]
Thanks @jmprieur we are really close :-)

**Pull request merged**